File: app/src/main/java/me/manabu/activities/parents/BasicToolbarActivity.java
package me.manabu.activities.parents;

import android.support.annotation.IdRes;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

abstract public class BasicToolbarActivity extends AppCompatActivity {

    protected Toolbar toolbar;

    protected void setToolbar(@IdRes int toolbarId) {
        toolbar = (Toolbar) findViewById(toolbarId);
        setSupportActionBar(toolbar);
    }

    public Toolbar getToolbar() {
        return toolbar;
    }
}

File: app/src/main/java/me/manabu/activities/parents/BasicToolbarActivity.kt
package me.manabu.activities.parents

import android.support.annotation.IdRes
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar

abstract class BasicToolbarActivity : AppCompatActivity() {

    lateinit var toolbar: Toolbar
        private set

    protected fun setToolbar(@IdRes toolbarId: Int) {
        toolbar = findViewById(toolbarId) as Toolbar
        setSupportActionBar(toolbar)
    }
}
